packages/pluggable-widgets-tools/utils/migration.js

```diff
@@ -0,0 +1,212 @@
+    { name: "@react-native-firebase/messaging", version: "17.3.0", check: CheckType.MAJOR_MINOR },
+    {
+        name: "react-native-camera",
+        version: "3.40.0",
+        check: CheckType.MAJOR_MINOR,
+        patch: "react-native-camera+3.40.0.patch"
+    },
+    {
+        name: "react-native-gesture-handler",
+        version: "1.10.3",
+        check: CheckType.MAJOR_MINOR,
+        patch: "react-native-gesture-handler+1.10.3.patch"
+    },
+    { name: "react-native-image-picker", version: "5.0.1", check: CheckType.MAJOR },
+    { name: "react-native-maps", version: "0.31.1", check: CheckType.MAJOR_MINOR },
+    { name: "react-native-progress", version: "^5.0.0", check: CheckType.MAJOR },
+    { name: "react-native-push-notification", version: "8.1.1", check: CheckType.MAJOR_MINOR },
+    { name: "react-native-webview", version: "11.26.1", check: CheckType.MAJOR_MINOR }
+];
+const resolutionsOverrides = [
+    { name: "react", version: "18.2.0", check: CheckType.MAJOR_MINOR },
+    { name: "react-native", version: "0.70.7", check: CheckType.MINOR }
+];
+
+function extractVersions(version) {
+    return version
+        .replace(/^\D+/, "")
+        .split(".")
+        .map(stringValue => Number(stringValue));
+}
+
+async function question(question) {
+    const rl = createInterface({ input: process.stdin, output: process.stdout });
+    return new Promise(resolve =>
+        rl.question(yellow(question), answer => {
+            rl.close();
+            resolve(answer);
+        })
+    );
+}
+
+function getOutdatedDependencies(packageDependencies, listOfNewDependencies = dependencies) {
+    const createReturn = dep => ({
+        name: dep.name,
+        oldVersion: packageDependencies[dep.name],
+        newVersion: dep.version,
+        patch: dep.patch
+    });
+    return listOfNewDependencies
+        .filter(dep => !!packageDependencies[dep.name])
+        .map(dep => {
+            const version = extractVersions(packageDependencies[dep.name]);
+            switch (dep.check) {
+                case CheckType.MAJOR:
+                    if (version[0] < extractVersions(dep.version)[0]) {
+                        return createReturn(dep);
+                    }
+                    break;
+                case CheckType.MINOR:
+                    if (version[1] < extractVersions(dep.version)[1]) {
+                        return createReturn(dep);
+                    }
+                    break;
+                case CheckType.MAJOR_MINOR:
+                    if (
+                        version[0] < extractVersions(dep.version)[0] ||
+                        (version[0] === extractVersions(dep.version)[0] && version[1] < extractVersions(dep.version)[1])
+                    ) {
+                        return createReturn(dep);
+                    }
+                    break;
+                default:
+                    return undefined;
```

Comment:
This default case should never happen, maybe we can improve on this.

Reply:
Jelmer itself fixed ✨